sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -43,7 +42,10 @@ case class Project(projectList: Seq[NamedExpression], child: SparkPlan) extends
 
   protected override def doExecute(): RDD[InternalRow] = child.execute().mapPartitions { iter =>
     val reusableProjection = buildProjection()
-    iter.map(reusableProjection)
+    iter.map { row =>
+      accumulators("numTuples").asInstanceOf[Accumulator[Long]] += 1
```

Comment:
This is called for every row. Could you save `accumulators("numTuples").asInstanceOf[Accumulator[Long]]` as a variable and reuse it?